### PR TITLE
Feature/IO-751_ci_logic

### PR DESCRIPTION
**Pull request by elizaveta-seleznyova-idf**

### CODE REVIEW

File: .github/workflows/commit_release_github_release.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        run: |
+          touch secrets.json
+          echo "{" >> secrets.json
+          echo " \"apiKeyTRAKT\": \"${{secrets.API_KEY_TRAKT}}\"," >> secrets.json
+          echo " \"apiKeyOMDB\": \"${{secrets.API_KEY_OMDB}}\"," >> secrets.json
+          echo " \"apiKeyTMDB\": \"${{secrets.API_KEY_TMDB}}\"," >> secrets.json
+          echo " \"facebook_app_id\": \"${{secrets.FACEBOOK_APP_ID}}\"," >> secrets.json
+          echo " \"fb_login_protocol_scheme\": \"${{secrets.FACEBOOK_CLIENT_TOKEN}}\"," >> secrets.json
+          echo " \"facebook_client_token\": \"${{secrets.FB_LOGIN_PROTOCOL_SCHEME}}\" " >> secrets.json
+          echo "}" >> secrets.json
```
Comment (vladimir-hodaryonok-idf) [inline .github/workflows/commit_release_github_release.yaml:35]: up to u but u can save in secrets whole json and add it by one string

File: .github/workflows/commit_release_github_release.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+          cat secrets.json
```
Comment (vladimir-hodaryonok-idf) [inline .github/workflows/commit_release_github_release.yaml]: remove this row (c) Sergey

File: .github/workflows/commit_release_github_release.yaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      - name: DataDependencies
+        run: |
+          flutter pub get
+        working-directory: packages/data
+      - name: DomainDependencies
+        run: |
+          flutter pub get
+        working-directory: packages/domain
+      - name: Build APK
+        run: flutter build apk --release
```
Comment (vladimir-hodaryonok-idf) [inline .github/workflows/commit_release_github_release.yaml:50]: where is flavour?

File: .github/workflows/commit_release_github_release.yaml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+      - name: Release version push to Github release
+        uses: ncipollo/release-action@v1
```
Comment (vladimir-hodaryonok-idf) [inline .github/workflows/commit_release_github_release.yaml]: use latest version (c) Sergey 
hint: ncipollo/release-action@v1.11.1

File: .github/workflows/pr_debug_firebase_distribution.yaml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+      - name: DataDependencies
+        run: |
+          flutter pub get
+        working-directory: packages/data
+      - name: DomainDependencies
+        run: |
+          flutter pub get
+        working-directory: packages/domain
+      - name: Build APK
+        run: flutter build apk --debug
```
Comment (vladimir-hodaryonok-idf) [inline .github/workflows/pr_debug_firebase_distribution.yaml:51]: -//-

File: lib/firebase_options.dart [hunk 6]
```diff
@@ -0,0 +1,70 @@
+// File generated by FlutterFire CLI.
```
Comment (vladimir-hodaryonok-idf) [inline lib/firebase_options.dart]: gitignore?

File: android/app/build.gradle [hunk 7]
```diff
@@ -57,7 +57,7 @@ android {
         applicationId "com.example.movies_and_stars"
         // You can update the following values to match your application needs.
         // For more information, see: https://docs.flutter.dev/deployment/android#reviewing-the-build-configuration.
-        minSdkVersion flutter.minSdkVersion
+        minSdkVersion 19
```
Comment (vladimir-hodaryonok-idf) [inline android/app/build.gradle:60]: I recommend 21 api version cuz, I faced a problem: "App requires Multidex support" when I tried to write long command for graddle. But up to u

**Review by vladimir-hodaryonok-idf — APPROVED** [review]

File: .github/workflows/commit_release_github_release.yaml (hunk 1 above)
Comment (elizaveta-seleznyova-idf) [reply .github/workflows/commit_release_github_release.yaml:35]: Not good idea
I asked Sergey and we came to that case

File: .github/workflows/commit_release_github_release.yaml (hunk 2 above)
Comment (elizaveta-seleznyova-idf) [reply .github/workflows/commit_release_github_release.yaml]: ok

File: .github/workflows/commit_release_github_release.yaml (hunk 3 above)
Comment (elizaveta-seleznyova-idf) [reply .github/workflows/commit_release_github_release.yaml:50]: I don't have flavours

File: .github/workflows/commit_release_github_release.yaml (hunk 4 above)
Comment (elizaveta-seleznyova-idf) [reply .github/workflows/commit_release_github_release.yaml]: ok

File: .github/workflows/pr_debug_firebase_distribution.yaml (hunk 5 above)
Comment (elizaveta-seleznyova-idf) [reply .github/workflows/pr_debug_firebase_distribution.yaml:51]: -//-

File: .github/workflows/pr_debug_firebase_distribution.yaml [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+      - name: Debug version push to Firebase App Distribution
+        uses: wzieba/Firebase-Distribution-Github-Action@v1.3.5
+        with:
```
Comment (egor-slatvinskiy-idf) [inline .github/workflows/pr_debug_firebase_distribution.yaml]: update to the latest version

File: .github/workflows/pr_debug_firebase_distribution.yaml [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+          echo " \"facebook_app_id\": \"${{secrets.FACEBOOK_APP_ID}}\"," >> secrets.json
+          echo " \"fb_login_protocol_scheme\": \"${{secrets.FACEBOOK_CLIENT_TOKEN}}\"," >> secrets.json
+          echo " \"facebook_client_token\": \"${{secrets.FB_LOGIN_PROTOCOL_SCHEME}}\" " >> secrets.json
+          echo "}" >> secrets.json
+          cat secrets.json
+      - name: PresentationDependencies
+        run: |
+          flutter pub get
+          flutter pub run intl_utils:generate
+        working-directory: packages/presentation
```
Comment (egor-slatvinskiy-idf) [inline .github/workflows/pr_debug_firebase_distribution.yaml:41]: I recommend specifying the working directory after the name, it's more readable

**Review by egor-slatvinskiy-idf — APPROVED** [review]

**Review by alexei-kruk-idf — APPROVED** [review]

**Review by yuliya-klescheva-idf — APPROVED** [review]

File: .github/workflows/commit_release_github_release.yaml [hunk 10]
```diff
@@ -0,0 +1,58 @@
+name: build_commit_debug_f_d
```
Comment (sergei-mikhailovskii-idf) [inline .github/workflows/commit_release_github_release.yaml]: script name is debug_f_d, but job is release_apk_to_gr. rename smth

File: .github/workflows/commit_release_github_release.yaml [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+  build_release_send_to_github_release:
+    name: ReleaseAPKtoGR
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '11'
+      - uses: subosito/flutter-action@v2
```
Comment (sergei-mikhailovskii-idf) [inline .github/workflows/commit_release_github_release.yaml]: use latest version plz

File: .github/workflows/pr_debug_firebase_distribution.yaml [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+  build_debug_send_to_firebase_destribution:
+    name: DebugAPKtoFD
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '11'
+      - uses: subosito/flutter-action@v2
```
Comment (sergei-mikhailovskii-idf) [inline .github/workflows/pr_debug_firebase_distribution.yaml]: -//-

**Review by sergei-mikhailovskii-idf — CHANGES_REQUESTED** [review]

File: .github/workflows/commit_release_github_release.yaml (hunk 10 above)
Comment (elizaveta-seleznyova-idf) [reply .github/workflows/commit_release_github_release.yaml]: ooooops

Comment (elizaveta-seleznyova-idf) [reply .github/workflows/commit_release_github_release.yaml]: renamed

File: .github/workflows/commit_release_github_release.yaml (hunk 11 above)
Comment (elizaveta-seleznyova-idf) [reply .github/workflows/commit_release_github_release.yaml]: add last version

File: .github/workflows/pr_debug_firebase_distribution.yaml (hunk 12 above)
Comment (elizaveta-seleznyova-idf) [reply .github/workflows/pr_debug_firebase_distribution.yaml]: -//-

**Review by sergei-mikhailovskii-idf — APPROVED** [review]

**Pull request merged**